src/Workspaces/Core/Portable/Formatting/Engine/TokenStream.cs

```diff
@@ -56,17 +53,13 @@ public TokenStream(TreeData treeData, OptionSet optionSet, TextSpan spanToFormat
 
                 // use some heuristics to get initial size of list rather than blindly start from default size == 4
                 int sizeOfList = spanToFormat.Length / MagicTextLengthToTokensRatio;
-                _tokens = new List<SyntaxToken>(sizeOfList);
-                _tokens.AddRange(_treeData.GetApplicableTokens(spanToFormat));
+                _tokens = _treeData.GetApplicableTokens(spanToFormat).ToImmutableArrayOrEmpty();
+                this.AverageTokenLength = Math.Max(1, (_tokens[_tokens.Length - 1].Position - _tokens[0].Position) * 1.0f / _tokens.Length);
```

Comment:
can this be commented to explain what's going on?

Reply:
intead of multiplying by 1f, just cast to double.

Reply:
Addressed that